### PR TITLE
TechDocs etc update to Backstage roadmap

### PR DESCRIPTION
**Pull request by garyniemen**
#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by freben — APPROVED** [review]
Approved, and cleaned up the vale and linting warnings

File: docs/overview/roadmap.md [hunk 1]
```diff
@@ -104,7 +104,15 @@ the maintainers’ radar, with clear interest expressed by the community.
 - **Security Plan (and Strategy):** The purpose of the Security Strategy is to
   move another step along the path to maturing the platform, setting the
   expectations of any adopters from a security standpoint.
-- **Search GA:**.
+- **Search 1.0:** Fix the few remaining issues to get Backstage Search platform
```
Comment (emmaindal) [inline docs/overview/roadmap.md:107]: could we link to the "few remaining issues" here to help the reader?

File: docs/overview/roadmap.md [hunk 2]
```diff
@@ -104,7 +104,15 @@ the maintainers’ radar, with clear interest expressed by the community.
...
+  up to 1.0
+- **TechDocs Addon framework and Addons:** Addons are TechDocs features that are
+  added on top of the base docs like code experience. An example would be a
+  feature that showed comments on the page. We plan to add an Addon framework
+  and open source a selection of the Addons that we use internally at Spotify.
+  Further Addons can then be added by the Community.
+- **Composable Homepage 1.0:** Driving this to 1.0 and adding some composable
```
Comment (emmaindal) [inline docs/overview/roadmap.md:114]: what does "Driving this to 1.0" mean here?

**Review by emmaindal — COMMENTED** [review]
Nice! Just some couple of comments from my side!

File: docs/overview/roadmap.md (hunk 1 above)
Comment (emmaindal) [reply docs/overview/roadmap.md:107]: some alternatives depending on what its referring to 
All issues labeled with search: https://github.com/backstage/backstage/issues?q=is%3Aissue+is%3Aopen+label%3Asearch

Search milestones:
https://github.com/backstage/backstage/milestone/27
https://github.com/backstage/backstage/milestone/28

**Review by emmaindal — APPROVED** [review]
Awesome, looks good to me! I fixed prettier issues, but we still need DCO sign off on your commits @garyniemen! Maybe @freben can help again!

**Pull request merged**